plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/OpenShiftShutdownController.java

```diff
@@ -36,9 +41,20 @@ public OpenShiftServerBehaviour getBehavior() {
 	@Override
 	public void stop(boolean force) {
 		OpenShiftServerBehaviour behavior = getBehavior();
+		IServer server = behavior.getServer();
+		IDeploymentConfig dc = OpenShiftServerUtils.getDeploymentConfig(server);
+		boolean debugMode = OpenShiftLaunchController.DEBUG_MODE.equals(server.getMode());
+
 		behavior.setServerStopping();
+
 		try {
-			OpenShiftDebugUtils.get().terminateRemoteDebugger(behavior.getServer());
+			if(dc == null) {
+				//something is wrong, do as much as possible.
+				OpenShiftDebugUtils.get().terminateRemoteDebugger(behavior.getServer());
+			} else if(debugMode) {
+				DebuggingContext debugContext = OpenShiftDebugUtils.get().getDebuggingContext(dc);
+				OpenShiftLaunchController.stopDebugging(server, dc, debugContext, new NullProgressMonitor());
```

Comment:
Yes, this method should return instance of OpenShiftLaunchController, but it is not an earlier created and cached instance, it seems that a new instance of controller is created at each request. In this case since we know exactly the class and method, can we just use constructor?